commands/build.sh

```diff
@@ -23,16 +41,34 @@ if [[ "$(plugin_read_config NO_CACHE "false")" == "false" ]] ; then
   for line in $(plugin_read_list CACHE_FROM) ; do
     IFS=':' read -r -a tokens <<< "$line"
     service_name=${tokens[0]}
-    service_image=$(IFS=':'; echo "${tokens[*]:1}")
+    service_image=$(IFS=':'; echo "${tokens[*]:1:2}")
+    cache_from_group_name=$(IFS=':'; echo "${tokens[*]:3}")
+    if [[ -z "$cache_from_group_name" ]]; then
+      cache_from_group_name=":default:"
+    fi
+    # The variable with this name will hold an array of group names:
     cache_image_name="$(service_name_cache_from_var "$service_name")"
 
     if [[ -n ${!cache_image_name+x} ]]; then
-      continue # skipping since there's already a pulled cache image for this service
+      if [[ "$(named_array_values ${cache_image_name})" =~ "${cache_from_group_name}" ]]; then
```

Comment:
@richafrank Please can you remove double quotes from right-hand side of =~ (i.e "${cache_from_group_name}") to match as a regex rather than literally

Reply:
Yep, thanks, will do! (My linter caught this too, but I wanted to ensure the tests covered it first.)

Reply:
@richafrank I triggered another build still saw error. Please can we update it as below

 if [[ "$(named_array_values "${cache_image_name}")" =~ ${cache_from_group_name} ]]; then

This should resolve the error

Reply:
Added the change to resolve this error

Reply:
Thanks @nsuma8989 !